ttnn/cpp/ttnn/tensor/xtensor/partition.cpp

```diff
@@ -0,0 +1,168 @@
+#include "ttnn/tensor/xtensor/conversion_utils.hpp"
+
+namespace ttnn::experimental::xtensor {
+namespace {
+
+using ::tt::tt_metal::Tensor;
+
+template <typename T, std::size_t... Indices>
+auto vector_to_tuple_helper(const std::vector<T>& v, std::index_sequence<Indices...>) {
+    return std::make_tuple(v[Indices]...);
+}
+
+template <std::size_t N, typename T>
+auto vector_to_tuple(const std::vector<T>& buffer) {
+    TT_FATAL(buffer.size() >= N, "Buffer size {} is less than N {}", buffer.size(), N);
+    return vector_to_tuple_helper(buffer, std::make_index_sequence<N>());
+}
+
+template <typename T, int N>
+xt::xarray<T> concat_helper(const std::vector<xt::xarray<T>>& v, size_t axis = 0) {
+    constexpr int FIXED_N = N < 2 ? 2 : N;
+    TT_FATAL(N > 1, "Tuple size in concatenate must be greater than 1; got N: {}", N);
+    auto tuple = vector_to_tuple<FIXED_N>(v);
+    return xt::concatenate(std::move(tuple), axis);
+}
+
+template <class T, size_t... I>
+consteval auto create_array_impl(std::index_sequence<I...>) {
+    return std::array<xt::xarray<T> (*)(const std::vector<xt::xarray<T>>& v, size_t axis), sizeof...(I)>{
+        concat_helper<T, I>...};
+}
+
+template <class T, size_t max>
+consteval auto create_array() {
+    return create_array_impl<T>(std::make_index_sequence<max>());
+}
+
+}  // namespace
+
+template <typename T>
+std::vector<xt::xarray<T>> chunk(const xt::xarray<T>& xtensor, int num_chunks, int dim) {
+    TT_FATAL(num_chunks > 0, "num_chunks must be > 0; got num_chunks: {}", num_chunks);
+    TT_FATAL(
+        dim >= 0 && dim < xtensor.dimension(),
+        "invalid dimension index; got dim: {}, tensor dimension: {}",
+        dim,
+        xtensor.dimension());
+
+    const int size_along_dim = static_cast<int>(xtensor.shape()[dim]);
+    TT_FATAL(
+        num_chunks <= size_along_dim,
+        "num_chunks cannot exceed the size of the tensor along the given dimension; got num_chunks: {}, "
+        "size_along_dim: {}",
+        num_chunks,
+        size_along_dim);
+
+    if (num_chunks == 1) {
+        return {xtensor};
+    }
+
+    const int chunk_size = (size_along_dim + num_chunks - 1) / num_chunks;
+    int remaining_size = size_along_dim;
+
+    std::vector<xt::xarray<T>> chunks;
+    chunks.reserve(static_cast<std::size_t>(num_chunks));
+    int start = 0;
+    int end = 0;
+    for (int i = 0; i < num_chunks && end < size_along_dim; ++i) {
+        int current_chunk_size = std::min(chunk_size, remaining_size);
+        remaining_size -= current_chunk_size;
+        end = start + current_chunk_size;
+
+        // Build indices for slicing
+        xt::xstrided_slice_vector indices(xtensor.dimension(), xt::all());
+        indices[dim] = xt::range(start, end);
+
+        auto chunk_view = xt::strided_view(xtensor, indices);
+
+        // Construct xarray from the view
+        // This forces a copy of that slice into a new xarray
+        chunks.push_back(xt::xarray<T>(chunk_view));
```

Comment:
I think this is fine for initial impl but we should be careful here that this is returning copies of slices instead of views like torch.chunk does: https://pytorch.org/docs/stable/generated/torch.chunk.html

Reply:
+1, added a TODO